### PR TITLE
[deckhouse] feat: set clusterDomain from global.clusterConfiguration.clusterDomain

### PR DESCRIPTION
**Pull request by juev**
## Description

If the domain field is filled in in the cluster configuration, we need to use it in the work of the hook to determine the domain name.

## Why do we need it, and what problem does it solve?

Sometimes it can be quite difficult to determine a domain name based on dynamic variables. And if the cluster configuration is defined, we just use it.

## What is the expected result?

If the domain name is defined in the cluster configuration, it will be used in the `global.discovery.clusterDomain` field.

## Checklist
- [X] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse
type: feature
summary: set clusterDomain from global.clusterConfiguration.clusterDomain
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by miklezzzz — APPROVED** [review]

File: global-hooks/discovery/cluster_domain.go [hunk 1]
```diff
@@ -70,6 +70,20 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
...
+			ApiVersion: "v1",
+			Kind:       "Secret",
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"kube-system"},
+				},
+			},
+			NameSelector: &types.NameSelector{
+				MatchNames: []string{"d8-cluster-configuration"},
+			},
```
Comment (yalosev) [inline global-hooks/discovery/cluster_domain.go:84]:
```suggestion
			},
			ExecuteHookOnSynchronization: pointer.Bool(false),
			ExecuteHookOnEvents: pointer.Bool(false),
```

File: global-hooks/discovery/cluster_domain.go [hunk 2]
```diff
@@ -99,17 +113,25 @@ func applyClusterDomainFromDNSPodFilter(obj *unstructured.Unstructured) (go_hook
...
 func discoveryClusterDomain(input *go_hook.HookInput) error {
-	clusterDomainCoreCMSnap := input.Snapshots[clusterDomainCoreCMSnapName]
-	clusterDomainDNSPodsSnap := input.Snapshots[clusterDomainDNSPodsSnapName]
+	// We have a hook for handling clusterConfiguration.
+	// During the operation of this hook, there is a blocking check for filling in the clusterDomain field.
+	// So now we just need to check that the `clusterConfiguration` is present in the secrets.
+	// And if this is the case, then the `global.discovery.clusterDomain` will be filled.
+	currentConfig, ok := input.Snapshots["clusterConfiguration"]
+	if ok && len(currentConfig) > 0 {
+		return nil
```
Comment (yalosev) [inline global-hooks/discovery/cluster_domain.go:122]: add a comment please, that the `cluster_configuration.go` hook handles this situation

Comment (juev) [reply global-hooks/discovery/cluster_domain.go:125]: We already have comment:
```plain
	// We have a hook for handling clusterConfiguration.
	// During the operation of this hook, there is a blocking check for filling in the clusterDomain field.
	// So now we just need to check that the `clusterConfiguration` is present in the secrets.
	// And if this is the case, then the `global.discovery.clusterDomain` will be filled.
```

File: global-hooks/discovery/cluster_domain.go (hunk 1 above)
Comment (juev) [reply global-hooks/discovery/cluster_domain.go:85]: Completed

**Review by yalosev — APPROVED** [review]

**Pull request merged**